hack/uwm_setup.sh

```diff
@@ -0,0 +1,259 @@
+#!/bin/bash
+#
+#*******************************************************************************
+#  NOTE THAT THIS SCRIPT IS INTENDED FOR OPENSHIFT CONTAINER PLATFORM 4.6 (OR
+#  LATER) TEST ENVIRONMENTS ONLY. THE SCRIPT MAKES CHANGES TO OAUTH IDENTITY
+# PROVIDER CONFIGURATION, OPENSHIFT USER ACCOUNTS AND USER WORKLOAD MONITORING
+#CONFIGURATION. THE SCRIPT ALSO DEPLOYS A USER APPLICATION IN A NAMESPACE CALLED
+#  'ns1'. REVIEW THIS SCRIPT THOROUGHLY TO DETERMINE SUITABILITY FOR YOUR TEST
+#                                  ENVIRONMENT
+#*******************************************************************************
+#
+# This script automates user workload monitoring setup for newly deployed OpenShift Container Platform 4.6 (or later) test clusters.The script does the following:
+#
+# * Enables user workload monitoring
+# * Deploys a user application
+# * Configures HTPasswd identity provider, creates a secret and creates some users
+# * Applies roles to those users
+# * Adds a service monitor
+# * Creates an alerting rule
+# * Outputs cluster and Thanos URLs to stdout
+#
+# The script assumes that YAML files `example-app-alerting-rule.yaml`, `example-app-service-monitor.yaml` and `prometheus-example-app.yaml` are present in the current working directory along with the script itself.
+
+# Add a trap which removes temporary files on SIGHUP, SIGTERM, SIGKILL, SIGQUIT, SIGINT and EXIT. Commented out by default. Review for suitability in your enviornment before uncommenting:
+# trap 'rm ./tmp/*' SIGHUP SIGTERM SIGKILL SIGQUIT SIGINT EXIT
+
+if [ ! -d "./tmp" ]; then
```

Comment:
What about changing this to `TMP=$(mktemp -d)`? This way it won't pollute repo directory and we don't need `trap EXIT rm ./tmp/*`.

Reply:
Thank you Pawel for your review! This is a great suggestion which I have now implemented in the script.